File: PolicyManagement.Api/PolicyManagement.Models/Commercial/CommercialPolicyFormDataModel.cs
using PolicyManagement.Dtos.Common;
using PolicyManagement.Models.Common;
using PolicyManagement.Models.Customer;
using PolicyManagement.Models.Health;
using PolicyManagement.Models.Motor;
using System;
using System.Collections.Generic;
using PolicyManagement.Infrastructures.EntityFramework;

namespace PolicyManagement.Models.Commercial
{
    public class CommercialPolicyFormDataModel
    {
        public CommercialPolicyFormDataModel()
        {
            PaymentData = new List<PaymentFormDataModel>();
            Document = new List<DocumentModel>();
        }
        public int PolicyId { get; set; }
        public int? PreviousPolicyId { get; set; }
        public CustomerFormDataModel Customer { get; set; }
        public PolicyTermFormDataModel PolicyTerm { get; set; }
        public string CoverNoteNumber { get; set; }
        public string CoverNoteIssueDateString { get; set; }
        public DateTime? CoverNoteIssueDate { get; set; }
        public DateDto CoverNoteIssueDateDto { get; set; }
        public TpOdPolicyFormDataModel TpPolicy { get; set; }
        public TpOdPolicyFormDataModel OdPolicy { get; set; }
        public NominationFormDataModel Nomination { get; set; }
        public int InsuranceBranch { get; set; }
        public int NumberOfKiloMeterCovered { get; set; }
        public int ExtendedKiloMeterCovered { get; set; }
        public short FinanceBy { get; set; }
        public PreviousPolicyFormDataModel PreviousPolicy { get; set; }
        public InspectionFormDataModel InspectionData { get; set; }
        public List<PaymentFormDataModel> PaymentData { get; set; }
        public VehicleFormDataModel Vehicle { get; set; }
        public PolicySourceFormDataModel PolicySource { get; set; }
        public PremiumFormDataModel Premium { get; set; }
        public string BranchId { get; set; }
        public string VerticalCode { get; set; }
        public List<DocumentModel> Document { get; set; }
        public AddOnRiderModel AddOnRider { get; set; }
        public string ControlNumber { get; set; }
        public short RenewalCounter { get; set; }
        public short VerticalSegmentId { get; set; }
        public short VerticalId { get; set; }
        public string Vertical { get; set; }
        public string PreviousControlNumber { get; set; }
        public short PolicyStatusId { get; set; }
        public string PolicyStatus { get; set; }
        public int PolicyCancelReasonId { get; set; }
        public string PolicyCancelReason { get; set; }
        public string DataEntryStatus { get; set; }
        public string ReconStatus { get; set; }
        public short? IsReconDone { get; set; }
        public string CommissionStatus { get; set; }
        public short? IsCommissionReceived { get; set; }
        public short? POSCommMonthCycleId { get; set; }
        public short? IRDACommMonthCycleId { get; set; }
        public string PolicyStatusColor { get; set; }
        public string DataEntryStatusColor { get; set; }
        public string CommissionStatusColor { get; set; }
        public string ReconStatusColor { get; set; }
        public string PolicyCancelReasonColor { get; set; }
        public bool Condition1 { get; set; }
        public bool Condition2 { get; set; }
        public bool IsBlockAgent { get; set; }
        public bool IsChangeAgent { get; set; }
        public string AddOnSelected { get; set; }
        public bool IsVerified { get; set; }
        public string CreatedBy { get; set; }
        public string VerifiedBy { get; set; }
        public DateTime? CreatedTime { get; set; }
        public DateTime? VerifiedTime { get; set; }
        public string ModifiedBy { get; set; }
        public DateTime? ModifiedTime { get; set; }
        public bool? IsPreviousPolicyApplicable { get; set; }
        public bool? Flag2 { get; set; }
        public bool? Flag1 { get; set; }
        public string QcCreatedBy { get; set; }
        public DateTime? QcCreatedTime { get; set; }
        public List<InsuredPersonModel> InsuredPersonData { get; set; }
        public short GenderId { get; set; }
        public ProductPlanModel ProductPlan { get; set; }
        public short? Portabality { get; set; }
        public DateTime? ContinueStartDate { get; set; }
        public DateDto ContinueStartDateDTO { get; set; }
        public int? NumberOfChild { get; set; }
        public int? NumberOfAdult { get; set; }
        public int? TotalSumInsured { get; set; }
        public tblFireCoverage FireCoverage { get; set; }


    }
}

File: PolicyManagementAngular/src/app/app-entites/models/motor/commercial-policy-form-data-model.ts
import { IDateDto } from "src/app/app-entites/dtos/common/date-dto";
import { ICustomerInsuranceDetail } from "../../dtos/customer/customer-short-detail-dto";
import { IDocumentModel } from "../common/document-model";
import { ICustomerFormDataModel } from "../customer/customer-form-data-model";
import { IAddOnRiderModel } from "./add-on-rider-model";
import { IInspectionFormDataModel } from "./inspection-form-data-model";
import { IMarineFormDataModel } from "./marine-policy-form-data-model copy";
import { INominationFormDataModel } from "./nomination-form-data-model";
import { IPaymentFormDataModel } from "./payment-form-data-model";
import { IPolicySourceFormDataModel } from "./policy-source-form-data-model";
import { IPolicyTermFormDataModel } from "./policy-term-form-data-model";
import { IPremiumFormDataModel } from "./premium-form-data-model";
import { IPreviousPolicyFormDataModel } from "./previous-policy-form-data-model";
import { IProductPlanFormDataModel } from "./product-plan-form-data-model";
import { ITpOdPolicyFormDataModel } from "./tp-od-policy-form-data-model";
import { IVehicleFormDataModel } from "./vehicle-form-data-model";

export interface ICommercialPolicyFormDataModel {
    PolicyId:number;
    BranchId: string;
    VerticalCode: string;
    Customer: ICustomerFormDataModel;
    CoverNoteNumber: string;
    CoverNoteIssueDateString: string;
    CoverNoteIssueDateDto: IDateDto | null;
    PolicyTerm: IPolicyTermFormDataModel;
    TpPolicy: ITpOdPolicyFormDataModel;
    OdPolicy: ITpOdPolicyFormDataModel;
    InsuranceBranch: number; // not mapped in api
    NumberOfKiloMeterCovered: string;
    ExtendedKiloMeterCovered: string;
    FinanceBy: number;
    PreviousPolicy: IPreviousPolicyFormDataModel;
    InspectionData: IInspectionFormDataModel;
    PaymentData: IPaymentFormDataModel[];
    PolicySource: IPolicySourceFormDataModel;
    Premium: IPremiumFormDataModel;
    Document: any;
    ControlNumber: string;
    AddOnRider: IAddOnRiderModel;
    PreviousControlNumber: string;
    PolicyCancelReason: string;
    Vertical: string;
    VerticalId: number;
    VerticalSegmentId: number;
    ReconStatus: string;
    CommissionStatus: string;
    PolicyStatus: string;
    RenewalCounter: number;
    PolicyStatusId: number;
    DataEntryStatus: string;
    DataEntryStatusColor: string;
    PolicyStatusColor: string;
    CommissionStatusColor: string;
    ReconStatusColor: string;
    PolicyCancelReasonColor: string;
    Condition1: boolean;
    Condition2: boolean;
    IsBlockAgent: boolean;
    IsChangeAgent: boolean;
    IsPreviousPolicyApplicable:boolean;
    AddOnSelected: string;
    IsVerified:boolean;
    VerifiedBy?: string;
    VerifiedTime?: string;
    ModifiedBy?: string;
    ModifiedTime?: string;
    CreatedBy?: string;
    CreatedTime?: string;
    PreviousPolicyId:number;
    Flag1?:boolean
    Flag2?:boolean;
    InsuredPersonData?: ICustomerInsuranceDetail[];
    Vehicle : IVehicleFormDataModel;
    Nomination: INominationFormDataModel;
    ProductPlan : IProductPlanFormDataModel;
    Portabality: number | null;
    ContinueStartDate : any | null;
    ContinueStartDateDTO? :IDateDto | null;
    NumberOfChild:Number;
    NumberOfAdult:Number;
    TotalSumInsured:Number;
    FireCoverage:any;
    Marine?: IMarineFormDataModel;
    Misc?: IMiscFormDataModel
    Liability?: ILiabilityFormDataModel
    Enginnering?: IEnginneringFormDataModel
    Gmc?: IGmcDataModel

}

export interface IMiscFormDataModel {
    MiscRate: number
    Misc1: number,
    Misc2: number,
    Misc3: number,
    Misc4: number,
}


export interface ILiabilityFormDataModel {
    LiabilityTermId:number,
    TotalSumInsured: number,
    NoWorker: number,
    NatureCoverage: string,
    ExcessClause: string,
    RiskLocation: number,
    OtherInfo: string,
    RetroSpectiveDate: any,
    PolicyId?:number 
}

export interface IEnginneringFormDataModel {
    EnginneringTermId:number,
    TotalSumInsured: number,
    Rate: number,
    NatureofCoverage: string,
    RiskLocation: number,
    OtherInfo: string,
    PeriodDate: any,
    EnginneringId?:any 
    PolicyId?:number 
}


export interface IGmcDataModel {
    CoverageType: string,
    NoEmployee: number,
    GmcTermId: number,
    NoDependent: string,
    NoLife: number,
    Coverage: string,
    CooperateLimit: string,
    OtherInfo: string,
    PolicyId?:number 
    Covertype:number 
}